src/Container/DefinitionResolver.php

```diff
@@ -96,7 +96,9 @@ public function resolve(DefinitionInterface $definition)
      */
     private function resolveReference($value)
     {
-        if ($value instanceof ReferenceInterface) {
+        if (is_array($value)) {
+            return array_map([$this, 'resolveReference'], $value);
```

Comment:
Actually, this is already recursive.

`resolveReference` is calling `resolveReference` (for each array element), so my solution should work with any level of array nesting. There is no need for `array_walk_recursive` in this particular case.